scripts/release-deprecate.ts

```diff
@@ -0,0 +1,86 @@
+import { compare } from 'compare-versions';
+import { exec } from 'node:child_process';
+import { readFileSync, readdirSync } from 'node:fs';
+import { join } from 'node:path';
+
+const { log, error } = console;
+
+const deprecateTags = /next|pr|rc/;
+const { version: currentVersion } = JSON.parse(
+  readFileSync(join(process.cwd(), '/packages/fuels/package.json')).toString()
+);
+const deprecateVersions = process.env.DEPRECATE_VERSIONS === 'true';
+
+const getPublicPackages = () => {
+  const packagesDir = join(__dirname, '../packages');
+  const packages = readdirSync(packagesDir, { withFileTypes: true });
+  const packagesNames = packages.map((p) => {
+    try {
+      const packageContent = readFileSync(join(packagesDir, p.name, 'package.json'), 'utf8');
+      const packageJson = JSON.parse(packageContent.toString());
+      return packageJson.private ? null : packageJson.name;
+    } catch (err) {
+      return null;
+    }
+  });
+  return packagesNames.filter((p) => !!p);
+};
+
+const getVersionsToDeprecate = async (packageName: string) => {
+  const { versions } = await fetch(`https://registry.npmjs.org/${packageName}`).then((resp) =>
```

Comment:
I wonder if we should do a `try/catch` and throw an error here, given npm _may_ return a non-200 status for the job given that amount of packages we are checking for. Have you seen such situations in the past?

Reply:
I haven't seen this endpoint fail, the only place I have seen it was on the deprecate CLI where we're doing thousands of versions rather than 10 packages, and even then it's still pretty resiliant.